tululbot/utils.py

```diff
@@ -1,18 +1,38 @@
+from datetime import datetime
 import random
 
 import requests
 import yaml
 
 
+class TimeHelper:
+
+    @classmethod
+    def now():
+        return datetime.datetime.now()
+
+
 class QuoteEngine:
 
     def __init__(self):
-        self._quote_url = 'https://cdn.rawgit.com/tulul/tulul-quotes/master/quote.yaml'  # noqa
         # Note: rawgit does not have 100% uptime, but at
         # least they're not throttling us.
+        self._quote_url = 'https://cdn.rawgit.com/tulul/tulul-quotes/master/quote.yaml'  # noqa
+
+        # Why not using rawgit's development endpoint? Because
+        # they can't promise 100% uptime on the development endpoint.
+        # Meanwhile, production endpoint's uptime is better because
+        # it is served by MaxCDN
+        self._git_branch_check_url = 'https://api.github.com/repos/tulul/tulul-quotes/branches/master'  # noqa
 
         self._cache = []
 
+        # URI refresh per interval in seconds
+        self._refresh_interval = 5 * 60
+        # Dummy date. Must be old enough (just to trigger)
+        # the uri must be refreshed on the first call
+        self._last_refresh = datetime(2009, 1, 6, 15, 8, 24, 78915)
```

Comment:
I see. this is so you don't have to handle the first call as special call like `if self._last_refresh is None`. cool.